lib/sleet/branch.rb

```diff
@@ -6,7 +6,7 @@ def initialize(circle_ci_token:, github_user:, github_repo:, branch:)
       @circle_ci_token = circle_ci_token
       @github_user = github_user
       @github_repo = github_repo
-      @branch = branch
+      @branch = CGI.escape(branch)
```

Comment:
Docs confirm that the branch name should be URL-encoded:
https://circleci.com/docs/api/#recent-builds-for-a-single-project